pilot/proxy/envoy/discovery_test.go

```diff
@@ -556,6 +571,10 @@ func TestListenerDiscoverySidecar(t *testing.T) {
 			name: "egress-rule",
 			file: egressRule,
 		},
+		{
+			name: "egress-rule-tcp",
+			file: egressRuleTCP,
+		},
```

Comment:
Remove

Reply:
If I remove this, how should I test that listeners are created for egress tcp rule?

Reply:
One simple idea: add multiple rules as part of egress rule yaml. Not sure if that works in test suite. I can’t figure out the egress tcp listener in the golden

Reply:
@rshriram `TestListenerDiscoverySidecar` is written to handle only a single piece of configuration data - an egress rule, a weighted route rule, a fault route rule. In general, is there a rationale to reduce the number of golden files? Why shouldn't we add as many as we want?

A possible solution could be to add these lines:
```
if testCase.name == "egress-rule" {
        addConfig(registry, egressRuleTCP, t)
}```

Which would make the code more complicated, comparing with adding just a config and a bunch of golden files.

Reply:
@rshriram The same question ^^^ regarding clusters discovery with a TCP Egress Rule testing - why shouldn't we have a separate test/tests for TCP Egress rules and a bunch of related golden files? Currently, I do not know where I should a TCP Egress Rule - `TestClusterDiscoveryAllClusters`, `TestClusterDiscovery`, `TestClusterDiscoveryCircuitBreaker` or some other `TestClusterDiscovery*` function ?